pilot/pkg/networking/plugin/mixer/mixer.go

```diff
@@ -122,3 +128,69 @@ func buildMixerOutboundTCPFilter(env *model.Environment, node *model.Proxy) list
 	// TODO(mostrowski): implementation
 	return listener.Filter{}
 }
+
+// buildHTTPMixerFilterConfig builds a mixer HTTP filter config. Mixer filter uses outbound configuration by default
+// (forward attributes, but not invoke check calls)  ServiceInstances belong to the Node.
+func buildHTTPMixerFilterConfig(mesh *meshconfig.MeshConfig, role model.Proxy, nodeInstances []*model.ServiceInstance, outboundRoute bool, config model.IstioConfigStore) *mccpb.HttpClientConfig { // nolint: lll
+	transport := &mccpb.TransportConfig{
+		CheckCluster:  v1.MixerCheckClusterName,
+		ReportCluster: v1.MixerReportClusterName,
+	}
+
+	mxConfig := &mccpb.HttpClientConfig{
+		MixerAttributes: &mpb.Attributes{
+			Attributes: map[string]*mpb.Attributes_AttributeValue{},
+		},
+		ServiceConfigs: map[string]*mccpb.ServiceConfig{},
+		Transport:      transport,
+	}
+
+	var labels map[string]string
+	// Note: instances are all running on mode.Node named 'role'
+	// So instance labels are the workload / Node labels.
+	if len(nodeInstances) > 0 {
+		labels = nodeInstances[0].Labels
+		mxConfig.DefaultDestinationService = nodeInstances[0].Service.Hostname
+	}
+
+	if !outboundRoute {
+		// for outboundRoutes there are no default MixerAttributes
+		// specific MixerAttributes are in per route configuration.
+		v1.AddStandardNodeAttributes(mxConfig.MixerAttributes.Attributes, v1.AttrDestinationPrefix, role.IPAddress, role.ID, labels)
+	}
+
+	if role.Type == model.Sidecar && !outboundRoute {
+		// Don't forward mixer attributes to the app from inbound sidecar routes
+	} else {
+		mxConfig.ForwardAttributes = &mpb.Attributes{
+			Attributes: map[string]*mpb.Attributes_AttributeValue{},
+		}
+		v1.AddStandardNodeAttributes(mxConfig.ForwardAttributes.Attributes, v1.AttrSourcePrefix, role.IPAddress, role.ID, labels)
+	}
+
+	for _, instance := range nodeInstances {
+		mxConfig.ServiceConfigs[instance.Service.Hostname] = v1.ServiceConfig(instance.Service.Hostname, instance, config,
+			outboundRoute || mesh.DisablePolicyChecks, outboundRoute)
+	}
+
+	return mxConfig
+}
+
+// buildTCPMixerFilterConfig builds a TCP filter config for inbound requests.
+func buildTCPMixerFilterConfig(mesh *meshconfig.MeshConfig, role model.Proxy, instance *model.ServiceInstance) *mccpb.TcpClientConfig {
+	attrs := v1.StandardNodeAttributes(v1.AttrDestinationPrefix, role.IPAddress, role.ID, nil)
```

Comment:
Can you please copy the code here? I know its duplication, but we are removing v1alpha3 enmasse from v1. So I don't want these things to get lost.

Reply:
i did it but i still don't understand the rationale. if you copy the code over you still have identical v1 code, just now it's harder to tell it's identical. you still have the same dependency it's just now masked. 
only if you actually make any changes to the code (however small) does it make sense to actually copy imo.